src/UI/Internal/Text.elm

```diff
@@ -334,13 +338,32 @@ spanMapOptions applier (Span prop opt) =
     Span prop (applier opt)
 
 
-spanRenderEl : RenderConfig -> Span -> Element msg
-spanRenderEl cfg (Span { content, size } { color, oneLineEllipsis }) =
+spanRenderEl : RenderConfig -> TextOptions -> Span -> Element msg
+spanRenderEl cfg { overflow } (Span { content, size } { color }) =
     content
-        |> ifThenElse oneLineEllipsis (ellipsizedText cfg size) Element.text
+        |> renderText cfg overflow size
         |> List.singleton
         |> Element.paragraph
-            (attributes cfg size oneLineEllipsis color)
+            (attributes cfg size overflow color)
+
+
+renderText : RenderConfig -> TextOverflow -> TextSize -> String -> Element msg
+renderText cfg overflow size text =
+    case overflow of
+        Wrap ->
+            Element.text text
+
+        Ellipsize ->
+            ellipsizedText cfg size text
+
+        EllipsizeWithTooltip ->
+            ellipsizedTextWith cfg
```

Comment:
This function is only being used for the "ellipsize with tooltip" situation right? do you think it needs to be "generic"  like this instead of just be a specific function?

Reply:
it's also used inside `ellipsizedText`, which is just a shorthand because it's still exposed and used in another instance. The difference between the two is the `(String -> Html msg)` function and the style props

Reply:
oh, true, I haven't realized, but then isn't better do create also a function ellipsizableWithTooltipText and use it here? I mean just to keep the consistency between both scenarios 🤔